multiversx/mx-chain-go
# En 13697 bootstrap metrics

## Reasoning behind the pull request
- When `StatusMetricsHandler` creation was moved to `statusCoreComponents` all metrics became available since bootstrap, thus no more `errNodeStarting` on metrics related API calls, leading to improper graphics. This change was needed in order to display real time trie sync status on termui.
  
## Proposed changes
- extend `StatusMetricsHandler` interface with a new method `BootstrapMetrics`
- add a new implementation of this interface, `InitialStatusMetricsHandler` which should return `errNodeStarting` for all routes except for a new one `/node/bootstrapstatus`, which will return all the metrics available during bootstrap

## Testing procedure
- standard system test
- metrics work as expected, mainly during bootstrap(error "node is starting" on /node/status and others except /node/bootstrapstatus)
- during trie sync(start a new node connected to this testnet), call `/node/bootstrapstatus` and confirm that the data is updated. Also check if termui trie sync status still looks like this
![image](https://user-images.githubusercontent.com/34831323/213202718-47f3c478-0efd-43eb-b12f-815aaa2881a8.png)


## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
